src/packet/signature/config.rs

```diff
@@ -24,42 +22,185 @@ pub struct SignatureConfig {
+            version_specific: SignatureVersionSpecific::V3 { created, issuer },
+        }
+    }
+
+    pub fn v4(typ: SignatureType, pub_alg: PublicKeyAlgorithm, hash_alg: HashAlgorithm) -> Self {
+        Self {
+            version: SignatureVersion::V4,
+            typ,
+            pub_alg,
+            hash_alg,
+            unhashed_subpackets: vec![],
+            hashed_subpackets: vec![],
+            version_specific: SignatureVersionSpecific::V4,
+        }
+    }
+
+    pub fn v6(
+        typ: SignatureType,
+        pub_alg: PublicKeyAlgorithm,
+        hash_alg: HashAlgorithm,
+        salt: Vec<u8>,
+    ) -> Self {
+        Self {
+            version: SignatureVersion::V6,
+            typ,
+            pub_alg,
+            hash_alg,
+            unhashed_subpackets: Vec::new(),
+            hashed_subpackets: Vec::new(),
+            version_specific: SignatureVersionSpecific::V6 { salt },
+        }
+    }
+
+    /// Generate v6 signature salt with the appropriate length for `hash_alg`
+    /// https://www.rfc-editor.org/rfc/rfc9580.html#name-hash-algorithms
+    fn v6_salt_for<R: CryptoRng + Rng>(mut rng: R, hash_alg: HashAlgorithm) -> Result<Vec<u8>> {
+        let Some(salt_len) = hash_alg.salt_len() else {
+            bail!("Unknown v6 signature salt length for hash algorithm {hash_alg:?}");
+        };
+
+        let mut salt = vec![0; salt_len];
+        rng.fill_bytes(&mut salt);
+
+        Ok(salt)
+    }
+
+    pub fn v6_with_rng<R: CryptoRng + Rng>(
+        rng: R,
+        typ: SignatureType,
+        pub_alg: PublicKeyAlgorithm,
+        hash_alg: HashAlgorithm,
+    ) -> Result<Self> {
+        Ok(Self::v6(
+            typ,
+            pub_alg,
+            hash_alg,
+            Self::v6_salt_for(rng, hash_alg)?,
+        ))
+    }
+
     pub fn new_v4(
-        version: SignatureVersion,
         typ: SignatureType,
         pub_alg: PublicKeyAlgorithm,
         hash_alg: HashAlgorithm,
         hashed_subpackets: Vec<Subpacket>,
         unhashed_subpackets: Vec<Subpacket>,
     ) -> Self {
         SignatureConfig {
-            version,
+            version: SignatureVersion::V4,
             typ,
             pub_alg,
             hash_alg,
             hashed_subpackets,
             unhashed_subpackets,
-            issuer: None,
-            created: None,
+            version_specific: SignatureVersionSpecific::V4,
         }
     }
 
+    pub fn version_specific<R>(
+        rng: R,
+        version: SignatureVersion,
+        hash_alg: HashAlgorithm,
+    ) -> Result<SignatureVersionSpecific>
+    where
+        R: CryptoRng + Rng,
+    {
+        match version {
+            SignatureVersion::V6 => Ok(SignatureVersionSpecific::V6 {
+                salt: Self::v6_salt_for(rng, hash_alg)?,
+            }),
+            SignatureVersion::V4 => Ok(SignatureVersionSpecific::V4),
+            _ => bail!("Unsupported signature version {:version?}"),
+        }
+    }
+
+    pub fn new_v4_v6<R>(
```

Comment:
do we need this? it reads really odd, and with the more specific `v4` and `v6` methods it seems redundant

Reply:
a while ago i thought it might be useful to have as a convenience constructor.
but dropping it seems right, yes: 33e884c701599218cdc61fc0d5b50e37993335d0